lib/block_view.go

```diff
@@ -7039,11 +7142,253 @@ func (bav *UtxoView) _connectAcceptNFTBid(
+		}
+
+		// If an old bid exists, delete it.
+		if prevNFTBidEntry != nil {
+			bav._deleteNFTBidEntryMappings(prevNFTBidEntry)
+		}
+
+		// If the new bid has a non-zero amount, set it.
+		if txMeta.BidAmountNanos != 0 {
+			// Zero bids are not allowed, submitting a zero bid effectively withdraws a prior bid.
+			newBidEntry := &NFTBidEntry{
+				BidderPKID:     bidderPKID.PKID,
+				NFTPostHash:    txMeta.NFTPostHash,
+				SerialNumber:   txMeta.SerialNumber,
+				BidAmountNanos: txMeta.BidAmountNanos,
+			}
+			bav._setNFTBidEntryMappings(newBidEntry)
+		}
+
+		// Add an operation to the list at the end indicating we've connected an NFT bid.
+		utxoOpsForTxn = append(utxoOpsForTxn, &UtxoOperation{
+			Type:            OperationTypeNFTBid,
+			PrevNFTBidEntry: prevNFTBidEntry,
+		})
+
+		return totalInput, totalOutput, utxoOpsForTxn, nil
+	} else {
+		// For bids on Buy Now NFTs, we create a bid that will get deleted in the _helpConnectNFTSold logic. This allows
+		// us to reuse the code that handles the royalty payouts and NFT ownership changes that is used in
+		// _connectAcceptNFTBid.
+		newBidEntry := &NFTBidEntry{
+			BidderPKID:     bidderPKID.PKID,
+			NFTPostHash:    txMeta.NFTPostHash,
+			SerialNumber:   txMeta.SerialNumber,
+			BidAmountNanos: txMeta.BidAmountNanos,
+		}
+		bav._setNFTBidEntryMappings(newBidEntry)
+		// Okay here's where the fun happens. We are submitting a bid on a Buy Now enabled NFT. We create the bid then we call the
+		// _helpConnectNFTSold to handle the royalty payout logic and such.
+		totalInput, totalOutput, utxoOpsForTxn, err := bav._helpConnectNFTSold(HelpConnectNFTSoldStruct{
+			NFTPostHash: txMeta.NFTPostHash,
+			SerialNumber: txMeta.SerialNumber,
+			BidderPKID: bidderPKID.PKID,
+			BidAmountNanos: txMeta.BidAmountNanos,
+
+			BidderInputs: []*DeSoInput{},
+
+			BlockHeight: blockHeight,
+			Txn: txn,
+			TxHash: txHash,
+			VerifySignatures: verifySignatures,
+			IsNFTBid: true,
+			IsAcceptNFTBid: false,
+		})
+		if err != nil {
+			return 0, 0, nil, errors.Wrapf(err, "_connectNFTBid: ")
+		}
+		return totalInput, totalOutput, utxoOpsForTxn, nil
+	}
+
+}
+
+type HelpConnectNFTSoldStruct struct {
+	NFTPostHash    *BlockHash
+	SerialNumber   uint64
+	BidderPKID     *PKID
+	BidAmountNanos uint64
+	UnlockableText []byte
+
+	// When an NFT owner accepts a bid, they must specify the bidder's UTXO inputs they will lock up
+	// as payment for the purchase. This prevents the transaction from accidentally using UTXOs
+	// that are used by future transactions.
+	BidderInputs []*DeSoInput
+
+	BlockHeight      uint32
+	Txn              *MsgDeSoTxn
+	TxHash           *BlockHash
+	VerifySignatures bool
+
+	// These both shouldn't be true
+	IsAcceptNFTBid   bool
+	IsNFTBid         bool
+}
+
+func (bav *UtxoView) _helpConnectNFTSold(txMeta HelpConnectNFTSoldStruct) (
+	_totalInput uint64, _totalOutput uint64, _utxoOps []*UtxoOperation, _err error) {
+	if txMeta.IsAcceptNFTBid == txMeta.IsNFTBid {
```

Comment:
Nit: I wouldn't call this txMeta. I would call it args since we use txMeta to mean something different in all other contexts.

I would also make args.Type be the thing that you use instead of having two mutually exclusive bools, as mentioned in the comment above.